zerver/lib/email_notifications.py

```diff
@@ -138,7 +138,17 @@ def fix_plaintext_image_urls(content: str) -> str:
         # with a simple hyperlink.
         return re.sub(r"\[(\S*)\]\((\S*)\)", r"\2", content)
 
-    def build_message_payload(message: Message) -> Dict[str, str]:
+    def append_sender_to_message(plain: str, html: str, sender: str) -> Tuple[str, str]:
+        plain = "{}: {}".format(sender, plain)
+
+        sender_tag = "<b>{}</b>:".format(sender)
+        if html[0:3] == "<p>":
+            html = "<p>{} {}".format(sender_tag, html[3:])
+        else:
```

Comment:
I'd be happier if we used an actual library for doing HTML manipulation, rather than string hackery, for this manipulation.

I think it's possible for this change to, for example, result in our putting unescaped HTML present in a user `full_name` field into these emails, which is probably fine in most email clients, but clearly isn't our intent.  

I think it's likely to be sufficient to HTML-escape `sender` and then keep the rest of the code as-is; but I wanted to flag this.

Random sidenote: I'm pretty sure we don't handle /me messages correctly in this logic; probably something to correct in a follow-up.

Reply:
I have replaced the string manipulation with bs4. 

> Random sidenote: I'm pretty sure we don't handle /me messages correctly in this logic; probably something to correct in a follow-up.

The current implementation looks like this.

![Screenshot from 2019-07-12 19-12-50](https://user-images.githubusercontent.com/7190633/61141535-6af19d80-a4eb-11e9-906d-cdeeb173d5ac.png)

The old version was 

![Screenshot from 2019-07-12 19-12-30](https://user-images.githubusercontent.com/7190633/61141573-7a70e680-a4eb-11e9-9e6a-7730cde7c784.png)

Should we make it work like web app?